csharp/src/Ice/InputStream.cs

```diff
@@ -16,6 +16,13 @@ namespace ZeroC.Ice
     /// <param name="istr">The input stream to read from.</param>
     public delegate T InputStreamReader<T>(InputStream istr);
 
+    /// <summary>A delegate that reads a value from an input stream. The value contains a stream parameter that
+    /// will be received using the given socket stream.</summary>
+    /// <typeparam name="T">The type of the value to read.</typeparam>
+    /// <param name="istr">The input stream to read from.</param>
+    /// <param name="socketStream">The socket stream to receive the streamable data from.</param>
+    public delegate T InputStreamReaderWithStreamable<T>(InputStream istr, SocketStream socketStream);
```

Comment:
I don't understand why we need this delegate.

When we unmarshal an incoming request/response frame that contains a stream parameter, it seems we could simply use a regular InputStreamReader for the non-stream parameters, and then a separate arg to read the stream parameter.

Reply:
So basically the  generated input stream reader would read the non-stream parameters from the encapsulation and this result of this operation would be handed over the a second reader that would assign the value of the stream parameter? 

I tried this originally but generating this was more complicated. It's much simpler to sill generate a single reader that takes care of assigning both the parameters read from the encapsulation (with the given istr) and the stream parameter assigned using the stream socket (using the given socketStream)

And it's also consistent with the generated writers that also takes care of both (writes the encapsulation parameters and returns a lambda to encode the stream parameter).